### DISCUSSION TITLE
Got runtime error when getting keying_material

### DISCUSSION THREAD

**Discussion by zhoucengchao** [segevfiner/sslkeylog#11] [Q&A] [answered]
Hi,

I am testing the fantastic sslkeylog module. I am able to get things like clientRandom, serverRandom, but not the keying_material with an runtime error. I think the provided parameters may have problem, but I am not sure what need to provide.

>>> sslkeylog.export_keying_material(ssl_sock,256,'key expansion')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/labadmin/.local/lib/python3.8/site-packages/sslkeylog.py", line 121, in export_keying_material
    return _sslkeylog.export_keying_material(sock, length, label, context)
RuntimeError: SSL_export_keying_material() failed
>>> 

Any ideas?

**Comment by segevfiner** [comment, marked as answer]
OpenSSL doesn't allow using certain reserved labels with `SSL_export_keying_material`: https://github.com/openssl/openssl/blob/cac250755efd0c40cc6127a0e4baceb8d226c7e3/ssl/t1_enc.c#L697-L699
